hive-connector/src/main/java/org/apache/iotdb/hive/TsFileDeserializer.java

```diff
@@ -0,0 +1,149 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.hive;
+
+
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.io.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class TsFileDeserializer {
+  private static final Logger LOG = LoggerFactory.getLogger(TsFileDeserializer.class);
+
+  private List<Object> row;
+
+  /**
+   * Deserialize an TsFile record, recursing into its component fields and
+   * deserializing them as well.  Fields of the record are matched by name
+   * against fields in the Hive row.
+   *
+   * Because TsFile has some data types that Hive does not, these are converted
+   * during deserialization to types Hive will work with.
+   *
+   * @param columnNames List of columns Hive is expecting from record.
+   * @param columnTypes List of column types matched by index to names
+   * @param writable Instance of MapWritable to deserialize
+   * @return A list of objects suitable for Hive to work with further
+   * @throws TsFileSerDeException For any exception during deserialization
+   */
+  public Object deserialize(List<String> columnNames, List<TypeInfo> columnTypes,
+                            Writable writable, String tableName) throws TsFileSerDeException {
+    if(!(writable instanceof MapWritable)) {
+      throw new TsFileSerDeException("Expecting a MapWritable");
+    }
+
+    if(row == null || row.size() != columnNames.size()) {
+      row = new ArrayList<>(columnNames.size());
+    } else {
+      row.clear();
+    }
+    MapWritable mapWritable = (MapWritable) writable;
+    String deviceId = mapWritable.get(new Text("device_id")).toString();
+    if (!Objects.equals(deviceId.toLowerCase(), tableName)) {
+      return null;
+    }
+
+    LOG.debug("device_id: {}", mapWritable.get(new Text("device_id")).toString());
+    LOG.debug("time_stamp: {}" , mapWritable.get(new Text("time_stamp")));
+
+    for(int i = 0; i < columnNames.size(); i++) {
+      TypeInfo columnType = columnTypes.get(i);
+      String columnName = columnNames.get(i);
+      Writable data = mapWritable.get(new Text(columnName));
+      if (data == null || data instanceof NullWritable) {
+        row.add(null);
+        continue;
+      }
+      if (columnType.getCategory() != ObjectInspector.Category.PRIMITIVE) {
+        throw new TsFileSerDeException("Unknown TypeInfo: " + columnType.getCategory());
```

Comment:
Maybe you could add `logger.error` here so that users can know the exact error position. Same with other exceptions below.
```suggestion
        logger.error("Unknown TypeInfo: {}", columnType.getCategory());
```